mono/mini/debugger-agent.c

```diff
@@ -3188,6 +3190,44 @@ process_frame (StackFrameInfo *info, MonoContext *ctx, gpointer user_data)
 	return FALSE;
 }
 
+static gint32 isFixedSizeArray (MonoClassField *f)
+{
+	ERROR_DECL (error);
+	if (!CHECK_PROTOCOL_VERSION (2, 53) || f->type->type != MONO_TYPE_VALUETYPE) {
+		return 1;
+	}
+	MonoCustomAttrInfo *cinfo;
+	MonoCustomAttrEntry *attr;
+	int aindex;
+	gint32 ret = 1;
+	cinfo = mono_custom_attrs_from_field_checked (f->parent, f, error);
+	if (!is_ok (error))
+		return ret;
+	attr = NULL;
+	if (cinfo) {
+		for (aindex = 0; aindex < cinfo->num_attrs; ++aindex) {
+			MonoClass *ctor_class = cinfo->attrs [aindex].ctor->klass;
+			if (mono_class_has_parent (ctor_class,  mono_class_load_from_name (mono_defaults.corlib, "System.Runtime.CompilerServices", "FixedBufferAttribute"))) {
+				attr = &cinfo->attrs [aindex];
```

Comment:
The class needs to be cached as well, see the GENERATE_GET_CLASS_WITH_CACHE macro.